File: app/src/main/java/pl/orbitemobile/wspolnoty/activities/home/domain/HomeScreen.java
package pl.orbitemobile.wspolnoty.activities.home.domain;

import io.reactivex.Single;
import pl.orbitemobile.wspolnoty.data.RemoteRepository;
import pl.orbitemobile.wspolnoty.data.entities.Article;

public class HomeScreen {
    
    private RemoteRepository mRemoteRepository;
    
    public HomeScreen() {
        mRemoteRepository = new RemoteRepository();
    }
    
    public Single<Article[]> getRemoteArticles() {
        return mRemoteRepository.getArticles();
    }
}

File: app/src/main/java/pl/orbitemobile/wspolnoty/activities/home/domain/HomeScreen.kt
/*
 * Copyright (c) 2017. All Rights Reserved. Michal Jankowski orbitemobile.pl
 */

package pl.orbitemobile.wspolnoty.activities.home.domain

import io.reactivex.Single
import pl.orbitemobile.kotlinrx.fromIoToMainThread
import pl.orbitemobile.wspolnoty.data.RemoteRepositoryImpl
import pl.orbitemobile.wspolnoty.data.dto.ArticleDTO

class HomeScreen {

    private val remoteRepository: RemoteRepositoryImpl = RemoteRepositoryImpl.instance

    val remoteArticles: Single<Array<ArticleDTO>>
        get() = remoteRepository.getArticles().fromIoToMainThread()
}
